services/web/server/src/simcore_service_webserver/groups/_db.py

```diff
@@ -386,6 +387,9 @@ async def delete_user_in_group(
     )
 
 
-async def get_group_from_gid(conn: SAConnection, gid: GroupID) -> RowProxy | None:
-    res: ResultProxy = await conn.execute(groups.select().where(groups.c.gid == gid))
-    return await res.first()
+async def get_group_from_gid(conn: SAConnection, gid: GroupID) -> GroupAtDB | None:
+    row: ResultProxy = await conn.execute(groups.select().where(groups.c.gid == gid))
+    result = await row.first()
+    if result:
+        return GroupAtDB.from_orm(result)
+    return None
```

Comment:
Didn't notice this one you are right. But I will keep it like this for now, because it seems the code where this function is used is currently potentially expecting None and is working with it. If we will refactor the groups/organizations we can look into it.

Reply:
that is not what I meant. that function was added in this PR... so I guess by you :rofl: that you check for None now in the handler is also in this PR;)

Reply:
No it is not! :) I just refactored the output from `RowProxy | None` to `Group | None` and it was used in the `services/web/server/src/simcore_service_webserver/garbage_collector/_core_utils.py` where there is check and based on the None value, you jump to another loop in the for loop.

Reply:
Aa sorry and the function itself I just added a "service" layer in between, so we do not expose the DB layer in the handler.

Reply:
The fact that you do not return the RowProxy that is perfect. 
What I meant is the GroupNotFoundError, that is actually raised in the handler function, should have come directly from here. but no problem.